### PR TITLE
IRIS-4154 | Tracking for On Site Notifications

### PR DESCRIPTION
**Pull request by slayful**

### CODE REVIEW

File: extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.tracking.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		'jquery',
+		'wikia.log',
+		'wikia.tracker',
+		'ext.wikia.design-system.on-site-notifications.common'
+	], function ($, log, tracker, common) {
+		'use strict';
+
+		function Tracking(model) {
+			this.model = model;
+			this.labels = {
```
Comment (kvas-damian) [inline extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.tracking.js:11]: Does it makes sense to have labels dict when mapping to the same string?

File: extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.tracking.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+				}.bind(this));
+			},
+
+			notificationImpression: function (id) {
+				try {
+					log('impression - notification', log.levels.info, common.logTag);
+					var notification = this.findById(id);
+					this.trackImpression(this.toLabel(notification.type),
+						{value: this.booleanToGa(notification.isUnread)});
+				} catch (e) {
```
Comment (kvas-damian) [inline extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.tracking.js:41]: Is anything throwing an exception in this code block?

File: extensions/wikia/DesignSystem/scripts/spec/DesignSystemGlobalNavigationOnSiteNotifications.tracking.spec.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+					setVal: function () {
+					},
+					toString: function () {
+						return testCase.toStringOutput;
+					}
+				},
+				QueryString = function () {
+					return queryStringMock;
+				},
+				helper = modules['ext.createNewWiki.helper']({}, QueryString);
```
Comment (kvas-damian) [inline extensions/wikia/DesignSystem/scripts/spec/DesignSystemGlobalNavigationOnSiteNotifications.tracking.spec.js]: Are you sure you're testing the right module? ;)
`createWiki` doesn't seem to be related to OnSiteNotifications.

File: extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.tracking.js (hunk 1 above)
Comment (slayful) [reply extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.tracking.js:11]: TBH that was a copy-paste from Mercury. Fixing, so that the dicts are used properly where it makes sense (markAsRead and markAllAsRead).

File: extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.tracking.js (hunk 2 above)
Comment (slayful) [reply extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.tracking.js:41]: I just don't want any possible issues with tracking to blow up in the user's face.

File: extensions/wikia/DesignSystem/scripts/spec/DesignSystemGlobalNavigationOnSiteNotifications.tracking.spec.js (hunk 3 above)
Comment (slayful) [reply extensions/wikia/DesignSystem/scripts/spec/DesignSystemGlobalNavigationOnSiteNotifications.tracking.spec.js]: I ended up not doing any of this, removing. =)

**Review by kvas-damian — APPROVED** [review]
